### PR TITLE
Include simshow function

### PR DESCRIPTION
**Pull request by roflmaostc**
As discussed in #55 

Would be happy to receive feedback :)

Right now it basically displays only 2D arrays.

I think it would be nice to have the option for 3D arrays to be shown as RGB, if possible.

For other 3D arrays I think we can just error because `simshow` is not a fully fledged viewer but instead a simple viewer for inline displays of arrays

### CODE REVIEW

**Comment by johnnychen94** [comment]
@tlnagy do you happen to have some suggestion on this?

**Comment by roflmaostc** [comment]
Hi!
Just wanted to ping this again :)

**Comment by tlnagy** [comment]
Sorry for the delay @roflmaostc, but this looks a nifty and it works on my end:
image

It would be great if you add support for signed fixed point numbers (as discussed in #41 like @johnnychen94 mentioned), which currently have the awful behavior of:

image

I think something simple like subtyping on `Gray{<: Fixed}` and then centering the display at 0 (similar to the functions you already have implemented) would be fantastic.

**Comment by roflmaostc** [comment]
Thank, I'll try to look into it!

I probably remove the parameter `f` inside `simshow` because it's rather useless.

**Comment by roflmaostc** [comment]
@tlnagy do we introduce a new dependency  for the [FixedPointNumbers.jl](https://github.com/JuliaMath/FixedPointNumbers.jl)?

**Comment by johnnychen94** [comment]
FixedPointNumbers and Colors are always included as long as ImageCore is loaded https://github.com/JuliaImages/ImageCore.jl/blob/fc9ac05630ac07876963d412a295fade46e0094e/src/ImageCore.jl#L4-L6

**Comment by roflmaostc** [comment]
Would it be OK to convert the FixedPointNumbers to Float for displaying it?

Error such this occur:
```julia
rgumentError: FixedPointNumbers.Q0f7 is an 8-bit type representing 256 values from -1.0 to 0.992; cannot represent 1
```

What should I do in this case? Even the operation `arr = set_zero ? arr .- minimum(arr) : arr` can't be represented in FixedPointNumbers in some cases.

So think I can't recycle the other functionalities very well (how would a gamma work?).

Is something like this ok?
```julia
 function simshow(arr::AbstractArray{Colors.Gray{T}}) where {T<:Fixed}
     return simshow(Array{Gray{Float64}}(arr))
 end
```

File: src/simshow.jl [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    simshow(arr; γ=1)
+
+Displays a complex array. Color encodes phase, brightness encodes magnitude.
+Works within Jupyter and Pluto.
+
+# Keyword args
+* `γ` applies a gamma correction to the magnitude
+"""
+function simshow(arr::AbstractArray{T};
+                 γ=one(T)) where (T<:Complex)
```
Comment (johnnychen94) [inline src/simshow.jl:55]: I think we should still accept the same set of keywords but do nothing for them.

```suggestion
function simshow(arr::AbstractArray{T};
                 γ=one(T), kwargs...) where (T<:Complex)
```

Similar suggestion to other methods.

**Review by johnnychen94 — COMMENTED** [review]
As a new function with no compatibility consideration, I'm fully supportive on merging this and giving it a try. But before merging I hope @tlnagy shares some comments on if this supports his usage (especially on how FixedPointNumbers are handled) -- I hope this function can be used by more people and not just one or two.

**Comment by roflmaostc** [comment]
What's the state of it?

I really use that every day and need to copy paste it all the time :laughing:

**Comment by johnnychen94** [comment]
I was waiting for more people involved because I'm not so active here these days; nevertheless, let's merge this and give it a try since it's a greenfield implementation.

**Pull request merged**